packages/features/webhooks/lib/factory/InstantMeetingBuilder.ts

```diff
@@ -0,0 +1,26 @@
+import { WebhookTriggerEvents } from "@calcom/prisma/enums";
+
+import type { InstantMeetingDTO } from "../dto/types";
+import type { WebhookPayload } from "./types";
+
+export class InstantMeetingBuilder {
+  canHandle(triggerEvent: WebhookTriggerEvents): boolean {
+    return triggerEvent === WebhookTriggerEvents.INSTANT_MEETING;
+  }
+
+  build(dto: InstantMeetingDTO): WebhookPayload {
+    return {
+      triggerEvent: dto.triggerEvent,
+      createdAt: dto.createdAt,
+      payload: {
```

Comment:
_triggerEvent_, _createdAt_, _payload_ are the standard props for all existing webhook events or any future one that we introduce.

I think we need a common fn to build that and feature specific webhook payload builders only decide what comes inside _payload_ prop

Reply:
Thing is to do this we'd likely need a wrapper that pretty much takes in triggerEvent, createdAt and payload, and just wrap around. I think the benefit here would be standardization, but it does seem like an extra function doing not a lot.
I'm 50/50 on this at the moment, as I see the pro of standardizing this and splitting responsibilities further such that the payload builders only build payload, and not wrap the final webhook structure. My initial thought is that if we add this, it slips somewhere between payload building and sending the webhook, maybe as an extra step (perhaps `WebhookEnvelopeFactory`??)

Is this along the lines of what you're thinking?

Reply:
We can revisit if necessary down the line!